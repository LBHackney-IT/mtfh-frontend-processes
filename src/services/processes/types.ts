export enum Trigger {
  CloseProcess = "CloseProcess",
  CancelProcess = "CancelProcess",
}
