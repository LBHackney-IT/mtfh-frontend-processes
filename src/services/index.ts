export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
