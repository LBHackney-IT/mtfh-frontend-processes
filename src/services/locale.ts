import reactHtmlParser from "react-html-parser";

import { AssetAddress } from "@mtfh/common/lib/api/asset/v1";

const locale = {
  backButton: "Back",
  bookAppointment: "Book Appointment",
  next: "Next",
  cancel: "Cancel",
  cancelProcess: "Cancel process",
  change: "Change",
  views: {
    soleToJoint: {
      title: "Sole tenant requests a joint tenure",
      steps: {
        selectTenant: "Select person",
        checkEligibility: "Eligibility Checks",
        supportingDocuments: "Supporting documents",
        finish: "Finish",
        breachOfTenancy: "Breach of tenure check",
        requestDocuments: "Request Documents",
        reviewDocuments: "Review Documents",
        submitCase: "Submit case",
      },
      actions: {
        reassignCase: "Reassign case",
        cancelProcess: "Cancel process",
        caseActivityHistory: "Case activity history",
      },
    },
    selectTenants: {
      selectTenantHint:
        "This person will be asked for proof of relationship e.g. marriage or civil partnership certificate.",
      selectTenantLabel: "Who do you want to add as a joint tenant?",
      addToTenureText: "If the person you want to add is not listed you must first",
      addToTenureLink: "add them to the tenure",
      noHouseholdMembersOver18:
        "There are no household members over the age of 18 associated to this.",
    },
    processesMenu: {
      selectProcessDescription: () =>
        reactHtmlParser(
          "<p>Start a process by selecting a process category below. <br /> Where a process is not yet supported in <strong>Manage My Home</strong> these links will open the appropriate interim Google form.</p>",
        ),
    },
    checkEligibility: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Passed automatic eligibilty checks",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    breachOfTenancy: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Eligibility checks passed",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    reviewDocuments: {
      passedChecks: "Eligibility checks passed",
      documentsRequested: "Documents requested",
      checkSupportingDocumentsAppointment:
        "I need to make an appointment to check supporting documents",
      appointmentScheduled: "Office appointment scheduled",
      seenPhotographicId:
        "I confirm that an outcome letter has been sent to the resident",
      seenSecondId:
        "I confirm I have seen a second form of ID (does not have to be photographic)",
      isNotInImmigrationControl:
        "I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996",
      seenProofOfRelationship:
        "I confirm I have seen proof of relationship to the existing tenant",
      incomingTenantLivingInProperty:
        "I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  title: "Processes",
  loadingText: "Submitting...",
  components: {
    startProcess: {
      buttonLabel: "Start process",
      thirdPartyHeading: "Sharing information with third parties",
    },
    entitySummary: {
      tenurePaymentRef: "Tenure payment ref",
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;

        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,

          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
    },
  },
};

export default locale;
