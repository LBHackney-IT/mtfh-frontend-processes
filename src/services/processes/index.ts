export * from "./sole-to-joint";
export * from "./change-of-name";
