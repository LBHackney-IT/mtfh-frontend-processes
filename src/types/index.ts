export type TargetType = "tenure" | "property" | "person";

export interface IStartProcess {
  thirdPartyCondition?: string;
  thirdPartyComponent?: React.FC;
  subHeading?: string;
  subComponent?: React.FC;
}

export interface IProcess {
  processName: string;
  title: string;
  targetType: TargetType;
  startProcess: IStartProcess;
  states: {
    [state: string]: {
      state: string;
      triggers: Record<string, string>;
    };
  };
}
