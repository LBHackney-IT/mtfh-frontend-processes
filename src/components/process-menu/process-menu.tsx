import { Link as RouterLink } from "react-router-dom";

import { menu } from "../../services";
import { TargetType } from "../../types";

import { useAsset } from "@mtfh/common/lib/api/asset/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import { Details, Link } from "@mtfh/common/lib/components";
import { useFeatureToggle } from "@mtfh/common/lib/hooks";

const LegacyMenu = ({ items }) => {
  return (
    <ul className="mtfh-processes-menu__list">
      {items.map((item, index) => (
        <li key={index}>
          <Link variant="link" href={item.link} isExternal>
            {item.label}
          </Link>
        </li>
      ))}
    </ul>
  );
};

const Menu = ({ items, id, targetType }) => {
  return (
    <ul className="mtfh-processes-menu__list">
      {items.map((item, index) => (
        <li key={index}>
          {item.processes ? (
            <Details title={item.label}>
              <ul className="mtfh-processes-menu__process-list">
                {item.processes.map((process, pIndex) => (
                  <li key={pIndex}>
                    <Link
                      variant="link"
                      as={RouterLink}
                      to={`${process.link}/${targetType}/${id}`}
                    >
                      {process.label}
                    </Link>
                  </li>
                ))}
              </ul>
            </Details>
          ) : (
            <Link variant="link" isExternal href={item.link}>
              {item.label}
            </Link>
          )}
        </li>
      ))}
    </ul>
  );
};

interface ProcessMenuProps {
  id: string;
  targetType: TargetType;
}

export const ProcessMenu = ({ id, targetType }: ProcessMenuProps) => {
  const hasEnhancedProcessMenu = useFeatureToggle("MMH.EnhancedProcessMenu");

  const { data: tenure } = useTenure(id);
  const { data: asset } = useAsset(tenure?.tenuredAsset?.id || null);
  const queryData = {
    tenure,
    asset,
  };

  const filteredMenuByEntityType = menu
    .map((item) => {
      let link = item.link;
      if (item.getPplQuery) {
        const query = item.getPplQuery(queryData);
        if (query.length > 0) {
          link = `${link}?${query}`;
        }
      }
      return {
        ...item,
        link,
        ...(item.processes && {
          processes: item.processes.filter(
            (process) => process.targetType === targetType,
          ),
        }),
      };
    })
    .filter(
      (item) =>
        !Array.isArray(item.processes) ||
        (Array.isArray(item.processes) && item.processes.length),
    );

  const legacyMenu = menu.map((item) => {
    let link = item.link;
    if (item.getPplQuery) {
      const query = item.getPplQuery(queryData);
      if (query.length > 0) {
        link = `${link}?${query}`;
      }
    }
    return {
      ...item,
      link,
    };
  });

  return hasEnhancedProcessMenu ? (
    <Menu items={filteredMenuByEntityType} id={id} targetType={targetType} />
  ) : (
    <LegacyMenu items={legacyMenu} />
  );
};
