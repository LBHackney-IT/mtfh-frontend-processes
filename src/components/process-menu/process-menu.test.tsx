import { mockActiveTenureV1, render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { menu, processes } from "../../services";
import { ProcessMenu } from "./process-menu";

import * as assetV1 from "@mtfh/common/lib/api/asset/v1/service";
import { Asset } from "@mtfh/common/lib/api/asset/v1/types";
import * as tenureV1 from "@mtfh/common/lib/api/tenure/v1/service";
import { Tenure } from "@mtfh/common/lib/api/tenure/v1/types";
import { $configuration } from "@mtfh/common/lib/configuration";
import { AxiosSWRResponse } from "@mtfh/common/lib/http";

test("it renders legacy process list", async () => {
  const { container } = render(
    <ProcessMenu id={mockActiveTenureV1.id} targetType="tenure" />,
  );
  expect(container).toMatchSnapshot();
});

describe("feature toggle on", () => {
  beforeEach(() => {
    const features = $configuration.getValue();
    $configuration.next({
      MMH: {
        ...features.MMH,
        featureToggles: { EnhancedProcessMenu: true },
      },
    });
  });

  test("it renders expanded process list and redirects to process on click", async () => {
    render(<ProcessMenu id={mockActiveTenureV1.id} targetType="tenure" />);
    const input = screen.getByText(menu[0].label);
    userEvent.click(input);

    const processLink = await screen.findByText(processes.soletojoint.title);

    expect(processLink).toBeInTheDocument();

    processLink.click();

    expect(window.location.pathname).toContain("/processes/soletojoint");
  });

  test("it opens an external link in a new tab", async () => {
    render(<ProcessMenu id={mockActiveTenureV1.id} targetType="tenure" />);

    const processLink = screen.getByText(menu[2].label);

    expect(processLink).toHaveAttribute("href", `${menu[2].link}`);
    expect(processLink).toHaveAttribute("target", "_blank");
  });

  test("it only shows soletojoint for the specified targetType tenure", async () => {
    render(<ProcessMenu id={mockActiveTenureV1.id} targetType="tenure" />);

    expect(screen.queryByText(processes.soletojoint.title)).toBeInTheDocument();
  });

  test("it does not show soletojoint for the specified targetType property", async () => {
    render(<ProcessMenu id={mockActiveTenureV1.id} targetType="property" />);

    expect(screen.queryByText(processes.soletojoint.title)).not.toBeInTheDocument();
  });

  describe("prepopulate query", () => {
    test("Request a Legal referral has the correct query", () => {
      const tenure = {
        householdMembers: [
          {
            fullName: "Test0 Test",
            isResponsible: false,
          },
          {
            fullName: "Test1 Test",
            isResponsible: true,
          },
          {
            fullName: "Test2 Test",
            isResponsible: true,
          },
        ],
        tenuredAsset: {
          propertyReference: "004567124",
        },
      } as Tenure;
      const asset = {
        assetAddress: {
          addressLine1: "10 Address Street",
          postCode: "E8 1DY",
        },
      } as Asset;

      jest.spyOn(tenureV1, "useTenure").mockReturnValue({
        data: tenure,
      } as AxiosSWRResponse<Tenure>);

      jest.spyOn(assetV1, "useAsset").mockReturnValue({
        data: asset,
      } as AxiosSWRResponse<Asset>);

      const { container } = render(
        <ProcessMenu id={mockActiveTenureV1.id} targetType="tenure" />,
      );
      expect(container).toMatchSnapshot();
    });

    test("Other tenancy changes has the correct query", () => {
      const tenure = {
        paymentReference: "2083027608",
        householdMembers: [
          {
            fullName: "Test1 Test",
            isResponsible: true,
          },
        ],
        tenuredAsset: {
          propertyReference: "004567124",
          fullAddress: "10 Address Street E8 1DY",
        },
      } as Tenure;

      jest.spyOn(tenureV1, "useTenure").mockReturnValue({
        data: tenure,
      } as AxiosSWRResponse<Tenure>);

      const { container } = render(
        <ProcessMenu id={mockActiveTenureV1.id} targetType="tenure" />,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
