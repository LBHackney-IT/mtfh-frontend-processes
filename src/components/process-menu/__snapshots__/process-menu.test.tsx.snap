// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature toggle on Change of name it doesn't render Change of name if its tenure is not Active & Secure 1`] = `
<div>
  <ul
    class="mtfh-processes-menu__list"
  >
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Other tenancy changes
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record a tenancy / audit homecheck
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140"
        rel="noopener noreferrer"
        target="_blank"
      >
        Find legal referral response
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Request a legal referral
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record notice served
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Rehousing
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Resident contact log
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report Japanese knotweed
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown tree
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown garden
      </a>
    </li>
  </ul>
</div>
`;

exports[`feature toggle on Change of name renders only if person has only 1 tenure and it is Active & Secure 1`] = `
<div>
  <ul
    class="mtfh-processes-menu__list"
  >
    <li>
      <details
        class="govuk-details lbh-details"
        data-module="govuk-details"
        data-testid="mtfh-details"
        id="mtfh-details"
      >
        <summary
          class="govuk-details__summary"
        >
          <span
            class="govuk-details__summary-text"
          >
            Change to a tenancy
          </span>
        </summary>
        <div
          class="govuk-details__text"
        >
          <ul
            class="mtfh-processes-menu__process-list"
          >
            <li>
              <a
                class="govuk-link lbh-link lbh-link--no-visited-state"
                href="/processes/changeofname/start/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
              >
                Change of Name
              </a>
            </li>
          </ul>
        </div>
      </details>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Other tenancy changes
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record a tenancy / audit homecheck
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140"
        rel="noopener noreferrer"
        target="_blank"
      >
        Find legal referral response
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Request a legal referral
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record notice served
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Rehousing
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Resident contact log
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report Japanese knotweed
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown tree
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown garden
      </a>
    </li>
  </ul>
</div>
`;

exports[`feature toggle on prepopulate query Other tenancy changes has the correct query 1`] = `
<div>
  <ul
    class="mtfh-processes-menu__list"
  >
    <li>
      <details
        class="govuk-details lbh-details"
        data-module="govuk-details"
        data-testid="mtfh-details"
        id="mtfh-details"
      >
        <summary
          class="govuk-details__summary"
        >
          <span
            class="govuk-details__summary-text"
          >
            Change to a tenancy
          </span>
        </summary>
        <div
          class="govuk-details__text"
        >
          <ul
            class="mtfh-processes-menu__process-list"
          >
            <li>
              <a
                class="govuk-link lbh-link lbh-link--no-visited-state"
                href="/processes/soletojoint/start/tenure/987b8498-7a38-4192-90e8-9d75c1365956"
              >
                Sole tenant requests a joint tenure
              </a>
            </li>
          </ul>
        </div>
      </details>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform?usp=pp_url&entry.1942863541=Test1%20Test&entry.1090481808=10%20Address%20Street%20E8%201DY&entry.321634337=004567124&entry.582205322=2083027608"
        rel="noopener noreferrer"
        target="_blank"
      >
        Other tenancy changes
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record a tenancy / audit homecheck
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140"
        rel="noopener noreferrer"
        target="_blank"
      >
        Find legal referral response
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform?usp=pp_url&entry.1699199605=Test1%20Test&entry.2086190845=10%20Address%20Street&entry.1956036523=E8%201DY&entry.1140527432=004567124"
        rel="noopener noreferrer"
        target="_blank"
      >
        Request a legal referral
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record notice served
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Rehousing
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Resident contact log
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report Japanese knotweed
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown tree
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown garden
      </a>
    </li>
  </ul>
</div>
`;

exports[`feature toggle on prepopulate query Request a Legal referral has the correct query 1`] = `
<div>
  <ul
    class="mtfh-processes-menu__list"
  >
    <li>
      <details
        class="govuk-details lbh-details"
        data-module="govuk-details"
        data-testid="mtfh-details"
        id="mtfh-details"
      >
        <summary
          class="govuk-details__summary"
        >
          <span
            class="govuk-details__summary-text"
          >
            Change to a tenancy
          </span>
        </summary>
        <div
          class="govuk-details__text"
        >
          <ul
            class="mtfh-processes-menu__process-list"
          >
            <li>
              <a
                class="govuk-link lbh-link lbh-link--no-visited-state"
                href="/processes/soletojoint/start/tenure/987b8498-7a38-4192-90e8-9d75c1365956"
              >
                Sole tenant requests a joint tenure
              </a>
            </li>
          </ul>
        </div>
      </details>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform?usp=pp_url&entry.1942863541=Test1%20Test&entry.1090481808=undefined&entry.321634337=004567124&entry.582205322=undefined"
        rel="noopener noreferrer"
        target="_blank"
      >
        Other tenancy changes
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record a tenancy / audit homecheck
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140"
        rel="noopener noreferrer"
        target="_blank"
      >
        Find legal referral response
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform?usp=pp_url&entry.1699199605=Test1%20Test,%20Test2%20Test&entry.2086190845=10%20Address%20Street&entry.1956036523=E8%201DY&entry.1140527432=004567124"
        rel="noopener noreferrer"
        target="_blank"
      >
        Request a legal referral
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record notice served
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Rehousing
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Resident contact log
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report Japanese knotweed
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown tree
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown garden
      </a>
    </li>
  </ul>
</div>
`;

exports[`it renders legacy process list 1`] = `
<div>
  <ul
    class="mtfh-processes-menu__list"
  >
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Sole to Joint
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Other tenancy changes
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record a tenancy / audit homecheck
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140"
        rel="noopener noreferrer"
        target="_blank"
      >
        Find legal referral response
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Request a legal referral
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Record notice served
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Rehousing
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform"
        rel="noopener noreferrer"
        target="_blank"
      >
        Resident contact log
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report Japanese knotweed
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown tree
      </a>
    </li>
    <li>
      <a
        class="govuk-link lbh-link"
        href="https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628"
        rel="noopener noreferrer"
        target="_blank"
      >
        Report overgrown garden
      </a>
    </li>
  </ul>
</div>
`;
